ocs_ci/utility/ibmcloud.py

```diff
@@ -203,3 +205,232 @@ def add_deployment_dependencies():
     ]
     for cr in wa_crs:
         run_cmd(f"oc apply -f {cr}")
+
+
+class IBMCloud(object):
+    """
+    Wrapper for Ibm cloud
+    """
+
+    def restart_nodes(self, nodes, timeout=900, wait=True):
+        """
+        Reboot the node on IBM Cloud.
+
+        Args:
+            nodes (list): The worker node instance
+
+        """
+        logger.info("restart_nodes")
+        provider_id = nodes[0].get()["spec"]["providerID"]
+        cluster_id = provider_id.split("/")[5]
+
+        cmd = f"ibmcloud ks workers --cluster {cluster_id} --output json"
+        out = run_cmd(cmd)
+        worker_nodes = json.loads(out)
+
+        if len(worker_nodes) > 0:
+            for node in worker_nodes:
+                cmd = f"ibmcloud ks worker reboot --cluster {cluster_id} --worker {node['id']} -f"
+                out = run_cmd(cmd)
+                logger.info(f"Node restart command output: {out}")
+
+    def attach_volume(self, volume, node):
+        """
+        Attach volume to node on IBM Cloud.
+
+        Args:
+            volume (str): volume id.
+            node (str): worker node id to attach.
+
+        """
+        logger.info(
+            f"attach_volumes:{node[0].get()['metadata']['labels']['failure-domain.beta.kubernetes.io/zone']}"
+        )
+        provider_id = node[0].get()["spec"]["providerID"]
+        cluster_id = provider_id.split("/")[5]
+        worker_id = node[0].get()["metadata"]["labels"][
+            "ibm-cloud.kubernetes.io/worker-id"
+        ]
+
+        logger.info(f"volume is : {volume}")
+
+        cmd = f"ibmcloud is volume {volume} --output json"
+        out = run_cmd(cmd)
+        out = json.loads(out)
+
+        if len(out["volume_attachments"]) == 0:
+            cmd = (
+                f"ibmcloud ks storage attachment  create --cluster {cluster_id} --worker {worker_id}"
+                f"  --volume {volume} --output json"
+            )
+            out = run_cmd(cmd)
+            out = json.loads(out)
+            logger.info(f"attachment command output: {out}")
+        else:
+            logger.info(f"volume is already attached to node: {out}")
+
+    def detach_volume(self, volume, node=None, delete_from_backend=True):
+        """
+        Detach volume to node on IBM Cloud.
+
+        Args:
+            volume (str): volume id.
+            node (str): worker node id to detach.
+            delete_from_backend (bool): True for deleting the disk from the
+                storage backend, False otherwise
+
+        """
+        provider_id = node[0].get()["spec"]["providerID"]
```

Comment:
Same as above in `attach_volume` method - `node` parameter is string (based on the docstring for this method)?

Reply:
changed